rts/Lua/LuaUnsyncedCtrl.cpp

```
@@ -2159,6 +2160,29 @@ int LuaUnsyncedCtrl::SetUnitNoMinimap(lua_State* L)
 	return 0;
 }
 
+/***
+ * @function Spring.SetMiniMapRotation
+ * @param rotation radians
+ * @return nil
+ */
+int LuaUnsyncedCtrl::SetMiniMapRotation(lua_State* L)
+{
+	if (CLuaHandle::GetHandleSynced(L))
+		return 0;
+
+	const float radians = luaL_checkfloat(L, 1);
+
+	if (radians < 0 || radians > 270*(math::PI/180))
```

Why not handle negative rotations here? I feel like this would be surprising behaviour as angles rotate.

```c
// Get the signed quadrant of the angle.
const float quad = radians / (math::PI / 2);

// Wrap it into range [0, 3]
// See https://stackoverflow.com/a/42131495/317135
const int rotation = static_cast<int>(std::round((quad % 4 + 4) % 4));
```

---

Oops, commented before [00a2142](https://github.com/beyond-all-reason/spring/pull/1966/commits/00a2142c5c9c302a1cfce90ffeb6d9f7c0de87df), but the problem still remains. It's ignoring input based on an arbitrary range.

---

I was thinking myself why not implement the entire thing to handle it all, but felt redundant (or so I believed), no problem, it's best to handle any case thrown instead of ignoring

---

It's part of a public API, IMO should be able to say:

```lua
count = 0;

--...

if button_pressed then
  count = count + 1;
  Spring.SetMiniMapRotation(count * PI / 2);
end
```

---

well originally I had a code that follows the cam:
```lua
function widget:Update()
    local _, roty, _ = spGetCamRot()  -- Camera rotation in radians
    
    -- Convert camera rotation to 0-3 (90° increments)
    local rotOption = math.floor((roty / (math.pi/2) + 0.5) % 4)
    
    local clampedRotOption = (rotOption % 2 == 0) and rotOption or (rotOption - 1)
    clampedRotOption = clampedRotOption % 4
    
    -- Get current minimap rotation and clamp it to 0/2 (0° or 180°) for current support
    local currentMiniRotRadians = spGetMiniRot()
    local currentClamped = math.floor((currentMiniRotRadians / math.pi) + 0.5) % 2 * 2
    
    -- Only update if needed
    if clampedRotOption == currentClamped then
        return
    end
    
    -- Convert to radians
    spSetMiniRot(clampedRotOption * (math.pi/2))
end
```
but in theory, we can just add it to the widget that allows ctrl+shift+o to flip camera

---

If you're accepting `radians` it should be treated as radians. Radians loop over [0, 2π).

Your widget code would become:
```lua
function widget:Update()
   local _, y, _ = spGetCamRot()
    spSetMiniRot(y);
end
```

Which feels objectively better to me.

The button press was just an example of code that looks like it should work but would not due to the mistreatment of the `radians` argument.

---

Yeah that's the easiest solution to follow the camera, but as it stands, BAR Default Minimap doesn't support 90 or 270-degree behavior it'll just squish the map background cuz of geometry stubbornness) hence why I need to clamp it to 0 and 180 until/if ever support for other options is introduced

---

Don't design engine interfaces around specific BAR widgets.

---

Wasn't talking about the engine, meant the widget to control auto rotation